services/blockchain-indexer/shared/indexer/accountBalanceIndex.js

```diff
@@ -80,18 +80,23 @@ const triggerAccountsBalanceUpdate = async () => {
 		config.set.accountBalanceUpdate.batchSize,
 	);
 
+	const numAddressesScheduled = addresses.length;
 	try {
 		// eslint-disable-next-line no-restricted-syntax
-		for (const address of addresses) {
+		while (addresses.length) {
+			const address = addresses.shift();
```

Comment:
```suggestion
			const address = addresses.pop();
```

Reply:
I preferred using `shift` over `pop` to ensure FIFO. With `pop` it'd be `LIFO` and when the update fails we keep pushing the last address to the end of the list. So, using `pop` would then query for the same address for which the update had failed.